core/src/main/scala/org/apache/spark/util/AccumulatorV2.scala

```diff
@@ -166,7 +166,7 @@ abstract class AccumulatorV2[IN, OUT] extends Serializable {
     if (atDriverSide) {
       if (!isRegistered) {
         throw new UnsupportedOperationException(
-          "Accumulator must be registered before send to executor")
+          s"Accumulator '${this.name.get}' must be registered before send to executor")
```

Comment:
name need not be defined - see `name.isDefined` below.

Reply:
Oh, right. Fixed